src/k8s/pkg/k8sd/pki/control_plane.go

```diff
@@ -44,11 +47,14 @@ func NewControlPlanePKI(opts ControlPlanePKIOpts) *ControlPlanePKI {
 		opts.Years = 1
 	}
 
+	userDefinedSANs := utils.GetExtraSANsFromString(opts.ExtraSANs)
+	userDefinedIpSANs, userDefinedDnsSANs := utils.SeparateSANs(userDefinedSANs)
+
 	return &ControlPlanePKI{
 		hostname:                  opts.Hostname,
 		years:                     opts.Years,
-		ipSANs:                    opts.IPSANs,
-		dnsSANs:                   opts.DNSSANs,
+		ipSANs:                    append(opts.IPSANs, userDefinedIpSANs...),
```

Comment:
consider combining these to a single extraSANs field, then we can split to ip and dns sans internally